osu.Game.Rulesets.Catch/UI/ICatchComboCounter.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Catch.UI
+{
+    /// <summary>
+    /// An interface providing a set of methods to update the combo counter.
+    /// </summary>
+    public interface ICatchComboCounter : IDrawable
```

Comment:
Was there some particular intent behind this interface? Right now it seems superfluous as it has a single implementation within the scope of this PR.

Reply:
Yes, the interface is going to be used for `DefaultComboCounter` as well.

Reply:
I think this is overengineering for one but I guess I can accept it for future usage in other rulesets. Not sure why this interface is catch-specific though; I don't think it really has to be.

Reply:
Also because I didn't want to think way more ahead and expose it "for later", but I guess it's fine to move this one out as `IComboCounter`

Reply:
Actually not sure about moving this one, as it asks for the combo colour of the object that was hit, which is catch-specific to my knowledge.

Reply:
left it for now because there's no better way to do this for now, and it's a catch-specific interface as is for the reason above. if there was a way for `GameplayBeatmap` to relay `On{New,Revert}Result` calls this could be entirely removed.